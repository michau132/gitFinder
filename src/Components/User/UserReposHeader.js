import React from 'react';
import PropTypes from 'prop-types';
import {
  FormControlLabel, Checkbox, Grid, TextField, Button, withStyles,
} from '@material-ui/core';

const styles = {
  formControl: {
    margin: 0,
    textAlign: 'center',
  },
};

const UserReposHeader = ({
  handleFilterRepos,
  openSelectedRepos,
  handleHideSelectedRepos,
  handleShowAllRepos,
  handleSelectAllRepos,
  classes,
  restStore: {
    isShowAllBtnDisabled,
    selectedReposAreEmpty,
    filterProjectsInput,
    allReposAreSelected,
  },
}) => (
  <Grid container justify="space-between" alignItems="center" spacing={8}>
    <Grid item xs={1} container justify="center">
      <FormControlLabel
        classes={{ root: classes.formControl }}
        control={(
          <Checkbox
            value="checkedI"
            color="primary"
          />
      )}
        label="Select all"
        labelPlacement="top"
        onChange={handleSelectAllRepos}
        checked={allReposAreSelected}
      />
    </Grid>
    <Grid item xs={7} container alignItems="center">
      <TextField
        type="search"
        label="filter projects"
        variant="outlined"
        value={filterProjectsInput}
        onChange={handleFilterRepos}
        fullWidth
      />
    </Grid>
    <Grid item xs={4} container alignItems="center">
      <Grid container justify="space-around" spacing={8}>
        <Button
          onClick={openSelectedRepos}
          disabled={selectedReposAreEmpty}
          variant="contained"
          color="primary"
        >
        open
        </Button>
        <Button
          onClick={handleHideSelectedRepos}
          disabled={selectedReposAreEmpty}
          color="primary"
          variant="contained"
        >
        hide
        </Button>
        <Button
          onClick={handleShowAllRepos}
          disabled={isShowAllBtnDisabled}
          variant="contained"
          color="primary"
        >
        show all
        </Button>
      </Grid>
    </Grid>
  </Grid>
);

UserReposHeader.propTypes = {
  handleFilterRepos: PropTypes.func.isRequired,
  handleShowAllRepos: PropTypes.func.isRequired,
  openSelectedRepos: PropTypes.func.isRequired,
  handleHideSelectedRepos: PropTypes.func.isRequired,
  handleSelectAllRepos: PropTypes.func.isRequired,
  restStore: PropTypes.shape({
    isShowAllBtnDisabled: PropTypes.bool.isRequired,
    selectedReposAreEmpty: PropTypes.bool.isRequired,
    filterProjectsInput: PropTypes.string.isRequired,
    allReposAreSelected: PropTypes.bool.isRequired,
  }).isRequired,
  classes: PropTypes.shape({}).isRequired,
};

export default withStyles(styles)(UserReposHeader);
