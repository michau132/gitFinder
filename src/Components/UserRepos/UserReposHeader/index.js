
import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';
import Button from '../../Button';

const UserReposHeader = ({
  setReposOnKeyUp,
  openSelectedRepos,
  hideSelectedRepos,
  showAllRepos,
  selectAllRepos,
  allReposAreShown,
  selectedReposAreEmpty,
  filterProjectsInput,
  allReposAreSelected,
}) => (
  <div className={styles.filterForm}>
    <div className={styles.checkAll}>
      <span>Select all</span>
      <input type="checkbox" onChange={selectAllRepos} checked={allReposAreSelected} />
    </div>
    <form className={styles.form}>
      <input className={styles.formText} value={filterProjectsInput} type="text" placeholder="filter projects" onChange={e => setReposOnKeyUp(e.target.value)} />
      <Button
        type="button"
        onClick={openSelectedRepos}
        disabled={selectedReposAreEmpty}
        text="open"
      />
      <Button
        type="button"
        onClick={hideSelectedRepos}
        disabled={selectedReposAreEmpty}
        text="hide"
      />
      <Button
        type="button"
        onClick={showAllRepos}
        disabled={allReposAreShown}
        text="show all"
      />
    </form>
  </div>
);

UserReposHeader.propTypes = {
  setReposOnKeyUp: PropTypes.func.isRequired,
  showAllRepos: PropTypes.func.isRequired,
  allReposAreShown: PropTypes.bool.isRequired,
  selectedReposAreEmpty: PropTypes.bool.isRequired,
  filterProjectsInput: PropTypes.string.isRequired,
  allReposAreSelected: PropTypes.bool.isRequired,
  openSelectedRepos: PropTypes.func.isRequired,
  hideSelectedRepos: PropTypes.func.isRequired,
  selectAllRepos: PropTypes.func.isRequired,
};

export default UserReposHeader;
