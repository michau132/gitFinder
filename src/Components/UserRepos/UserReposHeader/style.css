.filterForm {
  display: flex;
  margin-bottom: 15px;
}

.form {
  display: inline-block;
  width: calc(100% - 50px);
  display: flex;
  justify-content: space-between;
}

.formText {
  flex-grow: 1;
}

.form button {
  margin: 0 5px;
}

.checkAll {
  text-align: center;
}
