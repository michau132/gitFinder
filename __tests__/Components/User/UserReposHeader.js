import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserReposHeader from '../../../src/Components/User/UserReposHeader';

describe('testing UserReposHeader', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      handleFilterRepos: jest.fn(),
      openSelectedRepos: jest.fn(),
      handleHideSelectedRepos: jest.fn(),
      handleShowAllRepos: jest.fn(),
      handleSelectAllRepos: jest.fn(),
      restStore: {
        isShowAllBtnDisabled: true,
        selectedReposAreEmpty: true,
        filterProjectsInput: 'project',
        allReposAreSelected: false,
      },
    };
    wrapper = shallow(<UserReposHeader {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('calls onChange event on select all checkbox', () => {
    wrapper.find('.checkbox').simulate('change');
    expect(props.handleSelectAllRepos).toHaveBeenCalled();
  });

  test('calls onChange event on filterProjectsInput', () => {
    const filterInput = wrapper.find('.input');
    expect(filterInput.prop('value')).toEqual(props.restStore.filterProjectsInput);
    filterInput.simulate('change', { target: { value: 'ddds' } });
    expect(props.handleFilterRepos).toHaveBeenCalledWith({ target: { value: 'ddds' } });
  });

  describe('testing buttons', () => {
    test('calls event on openSelectedRepos', () => {
      const openButton = wrapper.find('.open');
      openButton.simulate('click');
      expect(props.openSelectedRepos).toHaveBeenCalled();
    });

    test('calls event on hideSelectedRepos', () => {
      const hideButton = wrapper.find('.hide');
      hideButton.simulate('click');
      expect(props.handleHideSelectedRepos).toHaveBeenCalled();
    });

    test('calls event on showAllRepos', () => {
      const showAllButton = wrapper.find('.showAll');
      showAllButton.simulate('click');
      expect(props.handleShowAllRepos).toHaveBeenCalled();
    });
  });
});
