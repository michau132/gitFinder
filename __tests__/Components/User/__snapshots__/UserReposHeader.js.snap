// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserReposHeader matching snapshot 1`] = `
<UserReposHeader
  allReposAreSelected={false}
  classes={
    Object {
      "formControl": "UserReposHeader-formControl-1",
    }
  }
  filterProjectsInput="project"
  hideSelectedRepos={[MockFunction]}
  isShowAllBtnDisabled={true}
  openSelectedRepos={[MockFunction]}
  selectAllRepos={[MockFunction]}
  selectedReposAreEmpty={true}
  setReposOnKeyUp={[MockFunction]}
  showAllRepos={[MockFunction]}
/>
`;
